Katana/SideEffect.swift

```diff
@@ -187,20 +211,21 @@ public protocol SideEffect: AnySideEffect {
    This approach is not suggested and should be used only in rare cases
    
    - parameter context: the context of the side effect
+   - returns: the side effect return value that will be used to parameterize the related promise
    - throws: if the logic has an error. The related promise will be rejected
    - seeAlso: https://github.com/malcommac/Hydra/#awaitasync
-  */
-  func sideEffect(_ context: SideEffectContext<StateType, Dependencies>) throws
+   */
+  func sideEffect(_ context: SideEffectContext<StateType, Dependencies>) throws -> ReturnValue
```

Comment:
I was wondering: does this force devs to write "return" every single time?
If so, is it possible to add some syntactic sugar that make so that when the side effect returns nothing (aka void) the side effect can be written exactly like now? It would be super useful also for the migration

Reply:
no, if you define the function in the implementation as:
`func sideEffect(_ context: SideEffectContext<StateType, Dependencies>) throws`
the compiler is smart enough to understand what's happening